wabt/src/main/java/com/dylibso/chicory/wabt/Wat2Wasm.java

```diff
@@ -30,30 +33,45 @@ public final class Wat2Wasm {
     private Wat2Wasm() {}
 
     public static byte[] parse(File file) {
+        try (InputStream is = new FileInputStream(file)) {
+            return parse(is, file.getName());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static byte[] parse(String wat) {
+        try (InputStream is = new ByteArrayInputStream(wat.getBytes(StandardCharsets.UTF_8))) {
+            return parse(is, "temp.wast");
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static byte[] parse(InputStream is, String fileName) {
         Module module = Module.builder(Wat2Wasm.class.getResourceAsStream("/wat2wasm")).build();
+
         try (ByteArrayOutputStream stdoutStream = new ByteArrayOutputStream();
                 ByteArrayOutputStream stderrStream = new ByteArrayOutputStream()) {
-            try (FileInputStream fis = new FileInputStream(file);
-                    FileSystem fs =
-                            Jimfs.newFileSystem(
-                                    Configuration.unix().toBuilder()
-                                            .setAttributeViews("unix")
-                                            .build())) {
-
-                var wasiOpts = WasiOptions.builder();
 
-                wasiOpts.withStdout(stdoutStream);
-                wasiOpts.withStderr(stdoutStream);
+            try (FileSystem fs =
```

Comment:
We could merge this try-with-resources into the above, then add an normal try block. So it would look like this:
```java
try (var stdoutStream = ...;
        var stderrStream = ...;
        var fs = ...) {
    try {
        ...
    } catch (WASMMachineException e) {
        ...
    }
}
```
This is just a suggestion, since this is a problem with the original code.

Reply:
I tried to, but unfortunately that's not possible because `stdoutStream` is used also in the inner catch block.